test/unit_test.c

```diff
@@ -1147,7 +1147,7 @@ static int request_test_handler(struct mg_connection *conn, void *cbdata)
 	          "Content-Type: text/plain\r\n\r\n");
 
 	for (i = 0; i < 20; i++) {
-		mg_printf(conn, "%s\r\n", i);
+		mg_printf(conn, "%d\r\n", i);
```

Comment:
This must be %x here anyway.

This part of the code is already converted to the "check" unit test framework: 
https://github.com/civetweb/civetweb/blob/master/test/public_server.c#L323